marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/jobs.sjs

```diff
@@ -98,7 +100,31 @@ function saveNewJob(job) {
         }
       }, config.JOBDATABASE)
     );
-    if (!jobDoc && throwErrorIfMissing) {
+    if (jobDoc) {
+      if (xdmp.castableAs(XS_NAMESPACE, "dateTime", jobDoc.job.timeStarted)) {
+        const timeStarted = xs.dateTime(jobDoc.job.timeStarted);
+        // Determine if related flow or have been updated
+        try {
+          const fullFlow = fn.head(hubUtils.invokeFunction(function () {
+                return Artifacts.getFullFlow(jobDoc.job.flow);
+              }, config.FINALDATABASE
+          ));
+          const allArtifacts = [fullFlow].concat(Object.values(fullFlow.steps));
+          jobDoc.job.flowOrStepsUpdatedSinceRun = allArtifacts.some((artifact) => {
+            if (xdmp.castableAs(XS_NAMESPACE, "dateTime", artifact.lastUpdated)) {
+              return xs.dateTime(artifact.lastUpdated).gt(timeStarted);
+            }
+          });
+        } catch (e) {
+          // the flow no longer exists
+          jobDoc.job.flowOrStepsUpdatedSinceRun = true;
+        }
+        if (xdmp.castableAs(XS_NAMESPACE, "dateTime", jobDoc.job.timeEnded)) {
+          // Calculate the duration
+          jobDoc.job.duration = xs.dateTime(jobDoc.job.timeEnded).subtract(timeStarted);
```

Comment:
This function is currently "get job", which just grabs the job doc as-is from the jobs database. But it's no longer that now - now it's "get job and some other data about the job that HC cares about". That means every client of getJobDocWithId is paying for the cost of calculating "updated since". 

Thus, I think that only the client that cares about this additional data should have this logic in it. Right now, JobController is using a getJob DS endpoint, but so is FlowRunnerImpl, which doesn't care about this data. 

New DS endpoints are cheap - since the story has a title of "See the details", then I think an expected DS endpoint here would be "getJobWithDetails", where the description of the endpoint can provide some clue as to what "details" means. 

That also means that my above comment about extracting a function would apply to the new DS module - i.e. jobs.sjs shouldn't be modified at all. And the unit tests below are still fine, but they should invoke the DS module instead.